### PR TITLE
feat: removed certain predicates from consent list

### PR DESCRIPTION
**Pull request by BelgianNoise**

### CODE REVIEW

**Comment by BelgianNoise** [comment]
closes https://github.com/digita-ai/workid/issues/67

**Comment by BelgianNoise** [comment]
Also filtered out doubles: eg:
`#hasEmail -> 'Email'`
`#email => 'Email'`

image

File: packages/ui-transfer-components/lib/components/consent-request.component.ts [hunk 1]
```diff
@@ -47,14 +22,24 @@ export class ConsentRequestComponent extends LitElement {
...
 
+    const hiddenPredicates = [
+      'https://schema.org/uri',
+      'http://www.w3.org/1999/02/22-rdf-syntax-ns#type',
+      'http://www.w3.org/2006/vcard/ns#value',
+    ];
+
+    const predicatesToShow = this.purpose.predicates.filter((pred) => !hiddenPredicates.includes(pred));
+    const translatedPredicates = predicatesToShow.map((pred) => predicateTranslations[pred] ?? pred);
+    const uniqueTranslatedPredicates = translatedPredicates.filter((val, index, self) => self.indexOf(val) === index);
```
Comment (woutermont) [inline packages/ui-transfer-components/lib/components/consent-request.component.ts:33]: This should also work and is cleaner. Could you test this?

```suggestion
    const uniqueTranslatedPredicates = Array.from(new Set(translatedPredicates));
```

File: packages/ui-transfer-components/lib/components/consent-result.component.ts [hunk 2]
```diff
@@ -46,14 +21,24 @@ export class ConsentResultComponent extends LitElement {
...
 
+    const hiddenPredicates = [
+      'https://schema.org/uri',
+      'http://www.w3.org/1999/02/22-rdf-syntax-ns#type',
+      'http://www.w3.org/2006/vcard/ns#value',
+    ];
+
+    const predicatesToShow = this.purpose.predicates.filter((pred) => !hiddenPredicates.includes(pred));
+    const translatedPredicates = predicatesToShow.map((pred) => predicateTranslations[pred] ?? pred);
+    const uniqueTranslatedPredicates = translatedPredicates.filter((val, index, self) => self.indexOf(val) === index);
```
Comment (woutermont) [inline packages/ui-transfer-components/lib/components/consent-result.component.ts:32]: This should also work and is cleaner. Could you test this?

```suggestion
    const uniqueTranslatedPredicates = Array.from(new Set(translatedPredicates));
```

**Review by woutermont — APPROVED** [review]

**Pull request merged**